### PR TITLE
Api changes for dimensions

### PR DESCRIPTION
**Pull request by pavanky**
- Renamed dim_type to dim_t
- Changed `af_array` from `size_t` to `void *`
- Changed `dim_t` from `int` to `size_t`
- Made sure all kernels in cuda and opencl are using `int` for performance

### CODE REVIEW

File: src/api/c/shift.cpp [hunk 1]
```diff
@@ -18,12 +18,12 @@ using af::dim4;
...
 
 template<typename T>
-static inline af_array shift(const af_array in, const af::dim4 &sdims)
+static inline af_array shift(const af_array in, const int sdims[4])
 {
     return getHandle(shift<T>(getArray<T>(in), sdims));
 }
 
-af_err af_shift(af_array *out, const af_array in, const af::dim4 &sdims)
+af_err af_shift(af_array *out, const af_array in, const int sdims[4])
```
Comment (umar456) [inline src/api/c/shift.cpp]: sdims should be a pointer. This is equivalent but it implies that it is passing all of the values.

Comment (pavanky) [inline src/api/c/shift.cpp]: It is passing all four values.

**Comment by umar456** [comment]
This should have been 2 pull requests

File: src/api/c/shift.cpp (hunk 1 above)
Comment (umar456) [inline src/api/c/shift.cpp]: No its still a pointer. It just looks like it is passing by value. The [4] is irrelevant.

**Pull request merged**